### ISSUE TITLE
Implement a QuickFolders Standard License 

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#187]
For funding future development and maintenance of the QuickFolders Add-on and distributing the load from a few "Pro Users" I will implement a three tier system going forward from Thunderbird 91, until QF is completely ported to a web extension and finally maintenance free.

- Pro License: no change, an annual license giving the extra features outlined [at the support site](https://quickfolders.org/premium.html).
- Standard License: Support the current feature set without additional restrictions. I also intend to remove the warning about quickMove / quickJump being a "Pro feature" so there will be less disruption during daily work with these important functions.
- Unfunded version: at the moment I am considering how to slightly restrict the feature set for beginners and casual users. One thing I definitely plan is restricting the number of tabs to a number e.g. up to 12 tabs. The choice of "unfunded" as opposed to "free" will highlight the difference as QF is still an open source, user-modifyable piece of software. The only restriction to prtoect my investment is that of using the [CC BY-ND 4.0](http://creativecommons.org/licenses/by-nd/4.0/legalcode) license which prohibits building competing products from my work (No Derivatives) but allows adapting it to your own personal needs.

**Transition Phase to 3-tier model**
The hard thing will be the transition phase, I want to tie this to using Thunderbird 91 forward.

**License Messages**
The wording of the "x is a premium feature" will be modified for the functions that are now also available to standard users:
![image](https://user-images.githubusercontent.com/6314674/126865586-fc36da3e-38e8-459e-8cb1-dc7308fc3f17.png)
Suggested new wording: "x" is a feature that is supported through licenses, if you find this useful you can get a standard license or a Pro license for extended features"
The QF Pro licensing screen will have to be modified too and be made more flexible, similar to SmartTemplates.
- if the user has a standard license (and it is not close to expiry)
![image](https://user-images.githubusercontent.com/6314674/126865724-08969737-2f5e-4618-bef1-e04412520393.png)
- if the license is close to expiry (currently within 30 days in SmartTemplates)
![image](https://user-images.githubusercontent.com/6314674/126865782-f70cb06f-42e7-4048-bd45-0fda28d7fe7f.png)
so there is always a choice for upgrading at the same time as extending the license.

the file "license.txt" needs to be updated to reflect of the existence of the new Standard license.

**Additional Features in Standard License**
The following features will be opened to standard license users (and not just be restricted to Pro Licensees)
- Backup/Restore:
  [Load Configuration...] will be unlocked for Standard Li.cense. When clicking [Save Configuration...] wording must be changed to "you can backup your configuration now but will need at least a QF standard license to use the loading feature". The user can always use the copy tabs / paste tabs functions on the Advanced tab if they need to transfer a setup to a different machine / profile.
- Tab specific properties / Advanced Properties. Some of the features on this page should be restricted to "free version" users:
![image](https://user-images.githubusercontent.com/6314674/126866027-f8209769-486a-41e5-b141-bc24cec36478.png)
which ones?
- Skip folder button: this is kind of an advanced feature but it is also very important. The Pro version also adds the configurable shortcut, which is going to remain. Should Skip be disabled in the "unfunded version" or should I add a sliding notification and let the unfunded version keep using it?
- Reading List: as this is an expensive feature to maintain it will be restricted to 6 entries for "free as in beer" users. This way it can still be evaluated and tested.
- Number of QuickFolders tabs - this should be restricted in the "unfunded version" as the most visible way and incentive to obtain a license. My initial idea is having a maximum of 12 working tabs for that. If an existing user has more tabs, they should see the excess tabs as disabled or somehow marked and they shouldn't be able to add more tabs (or those tabs could be disabled for clicking, but still available as information). Not quite sure how to pull this off in the transition phase as I expect some pushback, and no doubt there will be bad reviews following this move. So the question is how to do this in a compelling manner without disrupting a user's workflow too much.

**Pricing, Periodicity**
My initial price will be €5 / $5 for the first year and then €4 / $4 on renewals / extensions. I am still considering a permanent license but again it won't make sense unless QuickFolders is fully converted. On the one hand it is important and obviously rewarding to earn money and be invested into this project and it is certainly unreasonable to expect this to work (and be supported) forever with such a currently volatile and constantly evolving platform as Thunderbird / Mozilla (they increased the release rhythm to a new minor version push every 4 instead of 6 weeks), but on the other I hand I acknowledge that it's not ideal to "hire" software when we could own it. 

As the license validation currently has a expiry date "baked in" I might offer a higher priced life-time license that could be set forward to 70 years (I am somewhat doubtful of Thunderbird  or maybe even desktop PCs even existing at that stage), which could be 5 times the price of a standard license, likewise we could have this with a Pro license. Let's discuss.

**Comment by jaysmithgit** [comment]
To me, these types of ideas and changes seem reasonable.  As a heavy user (with a large number [10,000] of folders), I am in support of anything that encourages and rewards you, the developer, for continuing to do this hard work.  I am sure that you already know this, but the level of devotion you put into QF is much, much higher than some other developers do for their projects.  You should be rewarded for that and encouraged to continue.

I would suggest more difference between  the Unfunded vs Standard features, and perhaps between Standard vs Pro. In my opinion, there needs to be fairly significant benefits for an Unfunded user to upgrade to Standard.  For example, in the Unfunded, only 4 tabs instead of 12 you mentioned.  Also moving more features (not sure what to suggest) from Unfunded to Standard. If the user is getting value from these features, the user should be wiling to pay for the Standard.  

I suggest that $5/$4 for Standard is a little low simply because of the processing/payment costs. Perhaps $7/$5. (If a user does not perceive $7 or $5 to be justified for the benefit, they are not a serious user and not likely to pay any money for anything!)  I do recognize that there are parts of the world where $7 or $5 is a lot of money.  However, if somebody is using a tool as powerful as TB, hopefully they understand the benefit of making an investment in using highly productive tools such as QF.

As a Pro user myself, I would be fine with a 20% increase in the Pro cost. (I don't mean $20% as a limit, I just mean that I would not hesitate to pay more.)  QF IS a valuable tool!

I DO very much like the idea of "lifetime" subscription.  Or if not "lifetime", then maybe 3-5 years (with the understanding that there may be a point at which you cannot continue development for many different reasons).  For me, **the most important reason for a longer time** is not about saving money, it is that I don't want to have manage and deal with renewals, payments, etc. any more often than absolutely necessary.  Thinking about protecting your interests, while I would be happy with 5x cost for "lifetime", that is cheap and it would mean that after 5 years you will be doing a lot of work for free.  Be sure to protect your interests over the long term.

Related to this, though not directly part of what you asked, I should soon be paying for 3 or 4 Pro licenses (once I get other users fully converted to modern TB).  But, since COVID, we don't have 5+ people so the domain license does not make sense.  I am OKAY paying full price for the 3-4 licenses, BUT, if there is a way to put them all on the same schedule so I can renew them all (individually, I understand) at the same time, that would be VERY helpful. Maybe there is a way that you can make a "Small Domain" license that costs x times the number of users, but has a **SINGLE** point and time of payment each year.  That would be very attractive for me.

You asked some questions...

"Skip": I have no need for the feature and have never used it. Thus to me it is not important one way or the other, so I am not qualified to have an opinion about it.

"Reading List": Until you just asked for opinions, I did not know that this feature existed. It sounds useful and now I will think about how I can make use of it in my workflow.  I think this could be very restricted (1-3) for Unfunded users and somewhat restricted (4-6) for Standard users and unlimited for Pro.

"Number of QF tabs: Suggest 4 for Unfunded, 12 for Standard, unlimited for Pro.

To the extent that it is practical for you from a development standpoint, I would think that availability of features during the first 30 days -- for testing & demonstration purposes -- would be important.  In this way, the Unfunded user learns what is possible and what is useful for them, and then -- after 30 days -- they are more willing to pay money to continue to have the features that they have found especially useful.

These are all difficult decisions because I suspect that many users only use a small sub-set of features, so what is important to one user is of no use at all to another user.  :-)

Jay

**Comment by fredstluka** [comment]
Axel,

I agree with Jay on pretty much everything he said.  You should
be compensated for your time and effort.  Otherwise, you'll
eventually lose interest, and all of us Pro users will lose the value
you've been providing.  And without a revenue stream, you're
not likely to be able to find anyone else to take over for you.
So we'd all suffer the loss.

I'm happy to pay the $10/year that my Pro license currently costs,
and will probably continue to make my additional $100/year
annual donations.  But you'll have to judge your audience.  I'm
probably not a typical user, and I have no idea what price is
acceptable to most of your users and what price would start to
chase them off.

I agree that finding a way to demonstrate value with a free
(limited features) or trial (limited time) version is probably
a great way to get people to eventually buy a subscription.

Also, your YouTube videos are pretty compelling.  I'm not sure
whether any text description of the features could ever sell the
product as well as your videos do.

For me personally, the most useful features (really the only
features I use) are the shortcut keys for quickMove, quickCopy,
and quickJump.  I originally discovered QuickFolders a couple
of years ago, when I was doing Google searches to find a way
to increase the number of "recent" folders in native TB.  I was
tired of dropping down the Recent menu from the File toolbar
button, only to find that the folder I wanted had been replaced
by other more recent folders, and having to navigate the whole
tree of folders in the File toolbar button's dropdown, starting
from the account name.

Even though I am VERY much a user of keyboard and shortcuts,
almost never using the mouse if I can possibly avoid it, it hadn't
occurred to me how much easier it would be to just be able to
type a couple of chars to specify a folder like I now do with
quickMove.  QuickFolders has turned out to be an extraordinarily
good solution to a problem that I didn't even realize I had.  It
saves me huge amounts of time and aggravation every day.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

Just saw your private email on this topic.  Some answers to
your specific questions:

>   * restrictions for 'unfunded (as in "free") version'  users
>

Yes.  Add restrictions, as Jay suggested, to provide more
motivation for people to subscribe once they see the value.
That's a great way to get new subscribers.

But, it will inevitably annoy some of your existing free users.
Some will refuse to pay.   As Jay said, there are lots of people
who think all software should always be free of charge, not
just Open Source with freedom to use/modify/enhance.
Maybe you can afford to let those users go?

Or maybe you could find a way to "grandfather in" existing free
users?  Give them a temporary license for free (1 month, or 6
months, or 1 year, or something), but warn them that they'll
have to pay after that?  Be apologetic and appreciative, but tell
them honestly that you need to get some income or you'll have
to abandon the project.  Make them feel special, so they'll feel
loyal to you.

Maybe also offer them additional free license extensions or
upgrades to Pro if they refer additional paying customers to
you.  Get them to tell their friends about this great add-on
they've been using.  Post to their social media, etc.  Word of
mouth is a great tool, especially since all QuickFolders users
are probably thrilled with it.


>   * annual license (or tying to the Thunderbird version? this one is a
>     lot more difficult to implement though
>

Annual license.  There's no value in encouraging your users to
NOT upgrade to newer versions of Thunderbird.

>   * have a more expensive perpetual license (what pricing on that?),
>     the shop is going to get a whole lot more complicated :)
>

Yes, you could do that.  But, I'm not sure it's a good idea.
Users like Jay will love it, since they're already such fans of
QuickFolders, but more tentative users won't bother.  And
it will be a burden for you to have to support such users for
no additional income in future years.  And what if you had to
suddenly abandon the project, for some unavoidable reason?
Would users demand refunds?  If you do offer it, I'd expect it
to cost about 5X, like Jay said.


>   * If I introduce a permanent license for Pro users later, how would
>     I reward people who renewed an annual one for a number of years?
>

Don't worry about it.  They already got great value for the past
years.  Now they're about to get even better value by signing up
for a permanent license.  You could discount the permanent
license based on number of continuous years paid in the past,
but that would encourage more people to convert to permanent,
which I think is a bad idea anyhow.  What if everyone converted
to permanent?  You'd get a large cash influx, but zero future
income.  Eventually, that's going to be a problem.


>   * "what's new" text - I would like people to understand they can
>     roll back to an older version but that doesn't guarantee it will
>     be compatible with TB91 ESR.
>

I wouldn't even bother to mention it.  Those who don't want to
pay for even a standard license may guess that they can just
keep using an old version and that it may fail at some point as
they move to newer versions of TB.  I think you're trying WAY
too hard to keep everyone happy.  If you do mention it, just
saying they can "back to an older version but that doesn't
guarantee it will be compatible with TB91 ESR", is fine, I'd say.

Thoughts, Jay?

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by jaysmithgit** [comment]
Fred...

You have raised some excellent points and things for Axel to think about.  I don't have any specific further comments at this moment. Axel may just need to cogitate on this for a while and then perhaps make a specific near-final proposal upon which we and other users might make comments. Of course Axel must do what works best for him in both the short-term and long-term. However, these subjects are difficult because, at the outset, the long-term impacts cannot fully be known.

Jay

**Comment by RealRaven2000** [comment]
> I would suggest more difference between the Unfunded vs Standard features, and perhaps between Standard vs Pro. In my opinion, there needs to be fairly significant benefits for an Unfunded user to upgrade to Standard. For example, in the Unfunded, only 4 tabs instead of 12 you mentioned. Also moving more features (not sure what to suggest) from Unfunded to Standard. If the user is getting value from these features, the user should be wiling to pay for the Standard.

I know the argument but I do have data from my add-on SmartTemplates - I introduced a standard license and a 28 days trial period to transition users (not completely free / unfunded version) got a ton of bad reviews and lost 50% of the users within 6 weeks - there was a nag screen if you exceeded the trial period, and I suspect I get a lot of people who are just countend and annoyed by it - that's not the actual problem here though but the fact that it also seems to affect earnings / conversions to "pro license" users. The extension has been nothing but complicated work since then but I am still powering through. I think the idea is not to feel coerced by a piece of software. Since we do not have the option of "pay first then download" on the Mozilla Add-ons site, it's always something that needs to be told to the user after installation, which may look somewhat fraudulent for somebody who doesn't know the parameters that we as Add-on authors have to operate under.

> "Number of QF tabs: Suggest 4 for Unfunded, 12 for Standard, unlimited for Pro.

So I was thinking about the restriction of tabs, I believe 10 is probably the best compromise; I don't think somebody with a lot of folders who is really benefiting from QF will have this amount of less, and then I can still support the numbered shortcuts to try out for the majority of users. The question whether I should hide Categories (or restrict it to 3 categories) - it's actually problematic to completely take a way such a feature when users already use it - the users will just assume the product has become buggy.

> I suggest that $5/$4 for Standard is a little low simply because of the processing/payment costs. Perhaps $7/$5. (If a user does not perceive $7 or $5 to be justified for the benefit, they are not a serious user and not likely to pay any money for anything!) I do recognize that there are parts of the world where $7 or $5 is a lot of money. However, if somebody is using a tool as powerful as TB, hopefully they understand the benefit of making an investment in using highly productive tools such as QF.


Pricing is a difficult topic, I am hoping for more users if I keep it close to the "cup of coffee at the petrol station" level in order to make it annual, but I will probably also introduce a 5 years option for 20% off - which would at least remove a lot of the hassle of having regular payments - and then offering a permanent license as an upgrade or people could use it as a starting point to getting (extending into) the Pro License.


> As a Pro user myself, I would be fine with a 20% increase in the Pro cost. (I don't mean $20% as a limit, I just mean that I would not hesitate to pay more.) QF IS a valuable tool!

I was actually thinking of some increases along the lines of inflation, but I prefer lower prices with more users to make up for that. Marketing is hard though. If I treat my users well it's easier to grow an audience for it, generally shrinking is not such a good idea.

One of the reasons I keep the prices "oscillating" is that people love saving money so I do special offers around when I release new versions as you may have seen, the problem is that 10 different people have 10 different thresholds and ideas on what it a "fair price", so this has to be somehow serviced by the market.

> Related to this, though not directly part of what you asked, I should soon be paying for 3 or 4 Pro licenses (once I get other users fully converted to modern TB). But, since COVID, we don't have 5+ people so the domain license does not make sense. I am OKAY paying full price for the 3-4 licenses, BUT, if there is a way to put them all on the same schedule so I can renew them all (individually, I understand) at the same time, that would be VERY helpful. Maybe there is a way that you can make a "Small Domain" license that costs x times the number of users, but has a **SINGLE** point and time of payment each year. That would be very attractive for me.

I will need to look into it but I think I have to migrate all my product pages (16 of them, each with 9 different languages) from the classic interface to the new one first, because the current shop system isn't flexible enough to have something like a multiple email form and multiple license fulfillment.
![image](https://user-images.githubusercontent.com/6314674/127313470-7d7c21fd-d420-426e-aac2-8b4714177618.png)
I have already asked the question at fastspring support but I may have to try and get one of their representatives to assist (if that's what they do).

> "Skip": I have no need for the feature and have never used it. Thus to me it is not important one way or the other, so I am not qualified to have an opinion about it.

If you don't use `N` (next unread folder) then it won't be interesting but it is if you have a lot of unread folders and want to leave some for later (`Shift+S`), for my workflow it is essential.

> "Reading List": Until you just asked for opinions, I did not know that this feature existed. It sounds useful and now I will think about how I can make use of it in my workflow. I think this could be very restricted (1-3) for Unfunded users and somewhat restricted (4-6) for Standard users and unlimited for Pro.

Although I thought it is a great feature when I implemented it, I found that I wasn't using it an awful lot. That's mainly because it is another "list to worry about" and I try to not let anything accumulate if possible these days. So I am doubtful there is a big win in restricting it; it might even work to convince people against a license as they may view it as clutter.
 
> To the extent that it is practical for you from a development standpoint, I would think that availability of features during the first 30 days -- for testing & demonstration purposes -- would be important. In this way, the Unfunded user learns what is possible and what is useful for them, and then -- after 30 days -- they are more willing to pay money to continue to have the features that they have found especially useful.

I have something similar with SmartTemplates, but it is complicated to implement and didn't convert to added sales at all. I think to most users it's just a "complication" and they will just use the features much less or disable the Add-on. I think the black and white method of switching something off / on is probably easier to understand for most people. I haven't fully thought out how I am going to enforce the restriction to 10 tabs on a user who has more (should I hide the additional tabs, or disable them, and if so, in what ways?). quick brain storming on this:

- if the user tries to add more tabs they will get a dialog box saying "You need a standard license for more than 10 tabs"
- if the user clicks on an existing tab I could still jump to the folder but show a notification to the same end
- if they right-click or drag an email to the 11th (or higher) tab I could omit all subfolders and show a panel instead. This will make it possible for the user to "work around" the system temporarily by rearranging / dragging the most important tabs to the left, and make it very clear that they can contribute easily or take the restriction to save a very small amount of money.
- quickMove / quickJump - I could restrict the number of results to 8 items (and show a label - there are more results... which can show a message when clicked "You need a standard license for unlimited search results"). So again this will mainly affect "power users" who are already motivated to fund the Add-on.
- I would like to display a permanent reminder for people without license that can be displayed on (or instead of) the "QuickFolders" label, which could show instructions and jump to the license screen on hover:
![image](https://user-images.githubusercontent.com/6314674/127315416-ffbac5b0-0ad7-45d2-9fe7-c98b77832c3f.png)
![image](https://user-images.githubusercontent.com/6314674/127315600-77158a60-a1a8-49a5-ae58-4c18b31c0015.png)
I think this a lot better than a nag screen that just obstructs the work flow. Instead of showing it immediately, this could be on a 1 hour timer so it's not from the start of the session but appears eventually for people who work with email seriously.

**Comment by RealRaven2000** [comment]
> Yes.  Add restrictions, as Jay suggested, to provide more motivation for people to subscribe once they see the value. That's a great way to get new subscribers. But, it will inevitably annoy some of your existing free users. Some will refuse to pay.   As Jay said, there are lots of people who think all software should always be free of charge, not just Open Source with freedom to use/modify/enhance. Maybe you can afford to let those users go? 

Yes and no, I can see that even people who regularly renew licenses take breaks and you wouldn't necessarily want to lose a lot of these. If I keep the tool reasonably useful (while restriction) there is a bigger chance that they may come back to it quickly.

> Or maybe you could find a way to "grandfather in" existing free users?  Give them a temporary license for free (1 month, or 6 months, or 1 year, or something), but warn them that they'll have to pay after that?  Be apologetic and appreciative, but tell them honestly that you need to get some income or you'll have to abandon the project.  

I would be nice if I could automate a process for "trial licenses" like that but it's technically difficult as it relies on data I do not have. And it would require me to set it up as part of the fastspring shopping experience as I do not have a possibility to run server side script safely on my own (github hosted) pages. So it would be a big investment of my time and possibly cost, which is something I do not have in excess right now. IF people ask for free/ trial licenses personally (via email) I usually do this no questions asked, but this stuff is hard to "automate".

> * annual license (or tying to the Thunderbird version? this one is a lot more difficult to implement though
> Annual license.  There's no value in encouraging your users to NOT upgrade to newer versions of Thunderbird.

that's true. The main reason for that though it that it's a f..ton of work making my Add-ons compatible every year. Last time this education was done by not being available for Tb78 for 1 or 2 months but I think that's unprofessional, this time I want it to be seamless. See issue #114 

> * have a more expensive perpetual license (what pricing on that?), the shop is going to get a whole lot more complicated :)
> Yes, you could do that.  But, I'm not sure it's a good idea. Users like Jay will love it, since they're already such fans of QuickFolders, but more tentative users won't bother.  And it will be a burden for you to have to support such users for no additional income in future years.  And what if you had to suddenly abandon the project, for some unavoidable reason? Would users demand refunds?  If you do offer it, I'd expect it to cost about 5X, like Jay said.

I would have to offer it with a large T+C that says no guarantees for future versions of Thunderbird after the current ESR 91.* - My Add-ons will always _try_ to support one ESR [current + previous] (because whatever changes they implement, they will not care about whether current functionality is possible or maintained - it is a political act to convince them that we need Add-on toolbars for the future, and I have no guarantee of good leverage which we have at the moment with some Add-on friendly engineers in the team), which I would update very year to the next years version. This may be a deterrent to buy one but clarity is important.

> * If I introduce a permanent license for Pro users later, how would I reward people who renewed an annual one for a number of years?
> Don't worry about it.  They already got great value for the past years.  Now they're about to get even better value by signing up for a permanent license.  You could discount the permanent license based on number of continuous years paid in the past, but that would encourage more people to convert to permanent, which I think is a bad idea anyhow.  What if everyone converted to permanent?  You'd get a large cash influx, but zero future income.  Eventually, that's going to be a problem.

I know but if I can grow and get more users I might be able to make up for it. This would take marketing efforts but I might be able to finance them this way.

**Comment by fredstluka** [comment]
Axel,

Sounds like a tough nut to crack.  Dealing with a wide range
of budgets and opinions from a large number if users.  Trying
to not annoy any of them and cause your usage numbers to
drop.  I would find it very difficult.  Good luck with it!  In any
case, thanks for a great product!!

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
I am just working at the new rights management with the Standard License, already created [the product on the web shop](http://sites.fastspring.com/quickfolders/product/quickfoldersstandard ) (and an upgrade path from standard to pro, as well). I have rewritten the sliding notification that said about premium functions to be more flexible and made the necessary strings + translations [ `qf.notification.standard.readingList`  `qf.notification.standard.text`  `qf.notification.standard.title` 
`qf.licenseValidation.standard` ]. 

So here are my planned "feature access levels":

- Bookmarks: Unpaid = 5 max, Standard = 10 max, Pro = Unlimited
- Number of Tabs: Unpaid = 10 max, Standard + Pro unlimited. I might put a limit of 25 or so for standard users, or maybe restrict categories to 3 (new). I think 10 is a sweet spot for all the existing users and doesn't totally cripple them so they can still use it. I am "graying out" the 11th one and beyond, they can still be clicked but do not support subfolders (and I might disable drag+drop to them).
![image](https://user-images.githubusercontent.com/6314674/128424585-05fd50af-2e9f-40c0-a7e7-f381ec8ec090.png)
- paste folder entries (advanced tab) this was earmarked as Pro feature but I think it still worked. I will disable it for unpaid and convert it to a standard feature
- Save Config / Load Config: Load Config is Premium only, I might restrict Save Config (which is basically a backup) to Standard users at least.
- Skip unread folder: would like to restrict to Standard only, with a hint that the Pro feature has the keyboard shortcut
- quickMove / quickJump - the same here. It stands to reason that QuickFolders is still a useful tool without this so it's not completely crippled. They could potentially use Nostalgy (which currently hasn't got licenses) to make up for this, so it should make @opto a little happier as well. I wouldn't mind sharing (or even shedding) some users across to his Add-on.
- tab / folder icons - this is currently restricted to 12 without license, I might cut it down to 6 for unpaid users, 12 for standard and unlimited for "Pro" users (to do: create  a string `qf.notification.standard.tabIcons`)
- tab separators / line breaks. These may remain "Pro only" features
- Tab Specific Properties / Advanced Folder Properties Tab (I need to consolidate this function and it's menu entries, they are very inconsistent!): at the moment the function `updatePremiumFeatures()`  makes sure that the default identities and auto-"to:" settings of a folder cannot be applied recursively, I think this may also be a good restriction for Standard users (?)

Test version (you can already buy a standard license and upgrading _should_ work, too):
[QuickFolders-mx-5.7pre147.zip](https://github.com/RealRaven2000/QuickFolders/files/6941885/QuickFolders-mx-5.7pre147.zip)

**Comment by RealRaven2000** [comment]
I started creating a feature comparison chart here:

https://quickfolders.org/premium.html#featureComparison

**Comment by jaysmithgit** [comment]
Axel,

I think I need to get an "energy transfusion" from you. Your pace, energy, and creativeness are impressive. That could be a new revenue stream for you -- selling "pints of energy"!

1) Looks darn good to me.  Though "quickJump | .... Future versions may use a keyboard shortcut  for this."  :-(

Picky stuff...

2) In "Advanced Search settings" ("Here is a list of Premium features that are unrestricted to licensed users") section, there are some mismatches between the terminology used when talking about the features VS the actual text for those features in the illustrated dialog.  At first I was confused when reading the text and then trying to find that feature in the shown dialog image.

3) Typo: "single foldere match"

Since I have never taken much time to experiment with all the QF features beyond what I already needed when I came to Q, I actually learned quite a lot from this listing of features.

Here is a marketing idea for you... as if you need more work to do: Perhaps a monthly (weekly is fine with me, but that's more work for you) email [opt in] to your users that discusses a particular feature, how various users use that feature, possible workflows involving that feature, "this is what you could do with it" examples, etc. This might keep your users engaged and give them opportunities to recognize the benefits of upgrading to a paid level.

**Comment by RealRaven2000** [comment]
> I think I need to get an "energy transfusion" from you. Your pace, energy, and creativeness are impressive. That could be a new revenue stream for you -- selling "pints of energy"!

I wish I had any to spare. :)  I am just highly motivated by the August 10th deadline (Tb91 supposed to go live to 10% of users) - and I still have to complete a SmartTemplates promo (and new version), too. The idea is to never stop and just keep bringing the Add-ons working to avoid flattening the income curve.

> 1. Looks darn good to me.  Though "quickJump | .... Future versions may use a keyboard shortcut  for this."  :-(

Good catch! There are parts in this description that are outdated so it's always great to 

>  2. In "Advanced Search settings" ("Here is a list of Premium features that are unrestricted to licensed users") section, there are some mismatches between the terminology used when talking about the features VS the actual text for those features in the illustrated dialog.  At first I was confused when reading the text and then trying to find that feature in the shown dialog image.

I had to update the screenshot and description, I think these features evolved after I initially documented them.

> 3. Typo: "single foldere match"
> 
> Here is a marketing idea for you... as if you need more work to do: Perhaps a monthly (weekly is fine with me, but that's more work for you) email [opt in] to your users that discusses a particular feature, how various users use that feature, possible workflows involving that feature, "this is what you could do with it" examples, etc. This might keep your users engaged and give them opportunities to recognize the benefits of upgrading to a paid level.

Have been thinking about that one  but it's a lot of work. Then again I am very occasional with my YouTube channel because making a video is even more. I will keep that one in mind for the future. Fixed the errors and added on more item (and links) to the feature comparison chart, now I just need to link it from the UI (and translate at least the link into 19 languages).

**Comment by RealRaven2000** [comment]
Still working on the Splash screen with all the necessary info about the new standard license, it's a bit of a monster:

![image](https://user-images.githubusercontent.com/6314674/128555635-8c0d7497-98b4-49b5-a0c1-06a699a6ff39.png)

It's got some new links (one to the comparison chart, one directly to the standard license, and the license choice going to the internal registration dialog (which I also need to add the comparison link to). I will probably change the "New 2-tier Licensing System" for something less complicated. Thoughts?

**Comment by jaysmithgit** [comment]
These are really "picky" comments...

- The version I received via email and the version I see if I click on the one in the thread above are both larger than the one shown in the thread above.  Duh!  But, I mention this because in the smaller version it is hard to read that "outline reverse" "Future Proof..." etc. type font. That font only works at larger sizes.  As long as your viewers will be viewing at a larger size, then no problem.

- "Future Proof..." should at least be hyphenated "Future-Proof..." but some people may still not grasp the meaning. I am not sure what to suggests instead that still/continues to put it into the "benefit for the user" context.

- I don't know what "Dear thunderbird daily" means. Perhaps those who get that screen know what it means. At least "thunderbird" should be capitalized.

- Isn't the new license system really 3 tiers, not two? (free, basic, pro)? Or is the free not really considered a license.  But it is, isn't it?

- "As breaking changes..." might better be "Since [or Because] breaking changes..."

That's all I can think of to suggest, so it must be pretty good.

Jay

**Comment by RealRaven2000** [comment]
> These are really "picky" comments...

They are all very good points! It is so hard to be concise when you don't have an English Major...
 
> * The version I received via email and the version I see if I click on the one in the thread above are both larger than the one shown in the thread above.  Duh!  But, I mention this because in the smaller version it is hard to read that "outline reverse" "Future Proof..." etc. type font. That font only works at larger sizes.  As long as your viewers will be viewing at a larger size, then no problem.

I think this is just just the CSS on github that hits you - you can always zoom in on your browsers with `CTR [+]` .

> * "Future Proof..." should at least be hyphenated "Future-Proof..." but some people may still not grasp the meaning. I am not sure what to suggests instead that still/continues to put it into the "benefit for the user" context.

yeah this is a weak point, I want to come up with a better new slogan that encapsulates the dependencies of making changes to it keeps working in the next ESR. No idea how to put this into a tag line efficiently that can also be translated into other languages.

> * I don't know what "Dear thunderbird daily" means. Perhaps those who get that screen know what it means. At least "thunderbird" should be capitalized.

This one made me chuckle. It's actually dynamic and inserts the username (default identity), I just happened to screenshot from my thunderbird daily test profile.

> * Isn't the new license system really 3 tiers, not two? (free, basic, pro)? Or is the free not really considered a license.  But it is, isn't it?

It kind of is - except that you don't need a key for the free version. I think I am going to take the 2-tier out anyway, maybe, "Standard Licenses were Added" or such like. Haven't localized that one yet.

> * "As breaking changes..." might better be "Since [or Because] breaking changes..."

The whole sentence is clunky as heck, again I would like to find something that's a little softer, or drop that sentence.

**Comment by RealRaven2000** [comment]
Here is a first test version that implements some of the new "standard license" restrictions. The most important one being the restriction of the number of tabs: 

- maximum unrestricted tabs are 10 for unpaid and 25 for standard licenses
- already the existing tabs will be "grayed out" and will not accept dropped mail. You can still click them at the moment. You can also move tabs around so they are in the "front 10" / "front 25" to make them functional for the session.
- new tabs cannot be created if they exceed the new limits
- there will be some warning explaining the limitation with links to the web shop and license screen. The link to the feature comparison there is highlighted with "yellow marker" so it cannot be missed to easily.

[QuickFolders-mx-5.7pre208.zip](https://github.com/RealRaven2000/QuickFolders/files/6949498/QuickFolders-mx-5.7pre208.zip)

**Comment by RealRaven2000** [comment]
Implemented the remaining license restrictions. I also fixed the missing layout in the sliding notification bar (I had some trouble as it again exists in a "shadowRoot" element and I had to inject a special css file to have any influence on the contained elements in Thunderbird 91. We are just gnat's hair close to the release version!!!

- more specific messages for quickMove / quickJump / skip folder functions (added a hint about the shortcut with premium features)
- restriction of tab icons according to [feature comparison list](https://quickfolders.org/premium.html#featureComparison)
- tab separators / line breaks: these can only be removed even with standard license. only with pro license they can be added
- advanced folder properties: default from/to addresses (standard license) and recursive checkbox (pro license). when the "pro" button is added the layout is still super clunky in Tb91
- maximum reading list entries (unpaid: 5, standard: 10, pro: unlimited)

To install this preview, download the zip file and drag it into Thunderbird Add-ons manager.

[QuickFolders-mx-5.7pre254.zip](https://github.com/RealRaven2000/QuickFolders/files/6951603/QuickFolders-mx-5.7pre254.zip)
